Sources/Foundation/NSURL.swift

```diff
@@ -481,19 +481,24 @@ open class NSURL : NSObject, NSSecureCoding, NSCopying {
     
     // Memory leak. See https://github.com/apple/swift-corelibs-foundation/blob/master/Docs/Issues.md
     open var fileSystemRepresentation: UnsafePointer<Int8> {
-
 #if os(Windows)
-        let bufSize = Int(MAX_PATH + 1)
+        if let resolved = CFURLCopyAbsoluteURL(_cfObject),
```

Comment:
Can we add a comment here explaining why the Windows behavior needs to be different here but not in `_wideFileSystemRepresentation`?

Reply:
I suspect that the `_wideFileSystemRepresentation will need to change.  At that point, I'm more tempted to remove that function.  Having different variants of the functions makes things more difficult to correctly refactor.  That is, this is far more surgical than ideal.  I can add a note on the wide variant that it needs to be updated and ideally removed.

The honest answer is: fixing a bug at a time.  The reality is that most of the file system access routines for Windows need to be adjusted to use the NT path spelling, but that is also a good time to clean up some of the implementation to make it a bit less egregiously pessimistic (e.g. reusing encoded file paths, reusing stat information, etc).

Reply:
Discussed this offline - I'll add the comment in a follow up.  This is the first step towards fixing all the various cases in this file.  The work is going to enable DocC and should also help potentially help make SPM and driver more reliable on Windows with extended paths.